libcontainer/cgroups/fs/apply_raw.go

```diff
@@ -179,11 +180,24 @@ func (m *Manager) GetStats() (*cgroups.Stats, error) {
 }
 
 func (m *Manager) Set(container *configs.Config) error {
-	for name, path := range m.Paths {
-		sys, err := subsystems.Get(name)
-		if err == errSubsystemDoesNotExist || !cgroups.PathExists(path) {
+	for _, sys := range subsystems {
+		// We can't set this here, because after being applied, memcg doesn't
+		// allow a non-empty cgroup from having its limits changed.
+		if sys.Name() == "memory" {
```

Comment:
Yes. If you look at the old code, this was mentioned in `libcontainer/cgroups/fs/memory.go`. For some reason, the (outdated) kernel docs appear to suggest otherwise: https://www.kernel.org/doc/Documentation/cgroups/memory.txt (section 3.2). But on modern systems I've tested this on, the original memory code is correct and the kernel docs are wrong.